File: modules/module_02/src/dotprd.py
####################################################
# the following import is used to test the code
# ---------------------------------------------
#
import numpy as np
#
####################################################
####################################################
#
def dotprd(u,v):
    #
    # extract the number of components in the vector
    # to set the limits for the loop
    # ------------------------------
    #
    n = v.size
    #
    # create a temporary variable, sum, to tabulate
    # the results
    # -----------
    #
    sum = 0.0
    for i in range(n):
        sum = sum + u[i] * v[i]
    return sum
#
####################################################
####################################################
#
# The code beyond this point is not needed for the
# application above.
#
####################################################
n = 100
u = np.full(n, 1.0, dtype=float)
print(dotprd(u,u))
v = np.zeros(n)
sign = -1.0
fac = 1.0
for i in range(n):
    fac = fac * sign
    v[i] = fac * u[i] / float(i+1)
print(dotprd(v,u))
####################################################

File: modules/module_02/src/dotprd.c
////////////////////////////////////////////////////
// the following import is used to test the code
// ---------------------------------------------
////////////////////////////////////////////////////
#include <stdio.h>
#include <math.h>

double dotprod(double *a, &b);
//
////////////////////////////////////////////////////
//
double dotprd(*u,*v)
{
  //  
  // intitial the return sum to 0.0
  // ------------------------------
  double sum = 0.0;
  //
  // set the number of entries in the arrays
  // ---------------------------------------
  int n = sizeof(*u) / sizeof(*u[0]);
  //
  // loop over the entries, computing the component product, and then
  // adding this to the loop
  // -----------------------
  for(int i=0; i<n; i++) {
    sum = sum + u[i] * v[i];
  }
  //
  // return the dot product value
  // ----------------------------
  return sum;
  //
}
///////////////////////////////////////////////////
// The code beyond this point is not needed for the
// application above.
int main()
{
  int n = 100;
  double u[n];
  double v[n];
  double sigval = -1.0;
  for(int i=0; i++)
  {
    sigval = -1.0 * sigval;
    u[i] = 1.0;
    v[i] = sigval;
  }
  double dp = dotprd(u,v);
  printf("Dot Product Value: %g\n",dp);
}